python/tests/test_dict_encoding.py

```diff
@@ -404,3 +557,51 @@ def test_provenances(self):
             "provenances",
             ["record", "record_offset", "timestamp", "timestamp_offset"],
         )
+
+    def test_top_level_metadata(self):
+        tables = get_example_tables()
+        d = tables.asdict()
+        # None should give default value
+        d["metadata"] = None
+        lwt = c_module.LightweightTableCollection()
+        lwt.fromdict(d)
+        out = lwt.asdict()
+        self.assertNotIn("metadata", out)
+        tables = tskit.TableCollection.fromdict(out)
+        # We only removed the metadata, not the schema. So empty bytefield
+        # still gets interpreted
+        self.assertEqual(tables.metadata, {"top-level": []})
+        # Missing is tested in TestMissingData above
+
+    def test_top_level_metadata_schema(self):
+        tables = get_example_tables()
+        d = tables.asdict()
+        # None should give default value
+        d["metadata_schema"] = None
+        lwt = c_module.LightweightTableCollection()
+        lwt.fromdict(d)
+        out = lwt.asdict()
+        self.assertNotIn("metadata_schema", out)
+        tables = tskit.TableCollection.fromdict(out)
+        self.assertEqual(str(tables.metadata_schema), "")
+        # Missing is tested in TestMissingData above
+
+
+class TestExamples(unittest.TestCase):
+    def test_pickled_examples(self):
+        seen_current_msprime = False
+        test_dir = pathlib.Path(__file__).parent / "data/dict-encodings"
+        for filename in os.listdir(test_dir):
```

Comment:
Might as well use pathlib:
```
for filename in test_dir.glob("*.pkl"):
      # etc
```

Reply:
Fixed in https://github.com/tskit-dev/tskit/pull/687/commits/a8d83651c0842dacfbb4605ca067c82efbec1831